parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java

```diff
@@ -393,6 +396,16 @@ public ReadBuilder recordsPerBatch(int numRowsPerBatch) {
       return this;
     }
 
+    public ReadBuilder withNameMapping(NameMapping newNameMapping) {
+      this.nameMapping = newNameMapping;
+      return this;
+    }
+
+    public ReadBuilder applyNameMapping() {
```

Comment:
What is the purpose of this? Why not always apply a mapping if it was supplied?

I think it is confusing to only apply a mapping if this is called, even when the mapping is set.

Reply:
The idea is if there is no name mapping supplied, we use a default name mapping build from table schema.

I 'm still a bit confused that why we would allow users to pass in a customized name mapping. IMHO, there are only two name mappings should be legal, either we create it from parquet file or we create it from the table schema. Is there any scenario that user would pass in other mappings?

Reply:
I removed this and use name mapping from table property now.